src/vendor/api/v1/views.py

```diff
@@ -56,22 +71,28 @@ def post(self, request, *args, **kwargs):
 
         if not request.user.is_authenticated:
             request.session['session_cart'] = self.session_cart(request, offer)
-        else:
-            profile, created = self.request.user.customer_profile.get_or_create(site=get_site_from_request(request))
+            return redirect('vendor:cart')      # Redirect to cart on success
 
-            cart = profile.get_cart_or_checkout_cart()
+        profile, created = self.request.user.customer_profile.get_or_create(site=get_site_from_request(request))
 
-            if cart.status == InvoiceStatus.CHECKOUT:
-                cart.status = InvoiceStatus.CART
-                cart.save()
+        self.add_offer_to_customer_profile_cart(profile, offer)
+
+        return redirect('vendor:cart')      # Redirect to cart on success
+    
+    def get(self, request, *args, **kwargs):
+        try:
```

Comment:
I though about it, but I though it might look weird, and like this it is easier to see what is going on even if they are repeated. That's why I only created a different function for the part of adding it to the cart. 

In the future if we want to do somthing different in either it would be easier than wrapping everything. Don't know? What do you think?

Reply:
yeah, no worries! just wanted to see what you thought. we can keep it as is